### PR TITLE
docs: update README to include benchmarks

### PR DESCRIPTION
**Pull request by Arkenan**
Changes:

- Move spec tests to installing and running section, as that's where testing is.
- Make kurtosis an h2, to match the rest.
- Add benchmark section.
- Change order to installation - use - test - metrics and profiling - the rest (contributor package and other sections).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -280,55 +330,114 @@ make kurtosis.clean
...
+- `byte_reversal`: compares three different methods for byte reversal as a bitlist/bitvector operation. This concludes that using numbers as internal representation for those types would be the most efficient. If we ever need to improve them, that would be a good starting point. 
+- `shuffling_bench`: compares different methods for shuffling: shuffling a list in one go vs computing each shuffle one by one. Shuffling the full list was proved to be 10x faster.
+- `block_processing`: Builds a fork choice store with an anchor block and state. Uses the next block available to apply `on_block`, `on_attestation` and `on_attester_slashing` handlers. Runs these handlers 30 times. To run this, at least 2 blocks and a state must be available in the db. It also needs you to set the slot manually at the beginning of an epoch. Try it for the slot that appeared when you ran checkpoint sync (you'll see in the logs something along the lines of `[Checkpoint sync] Received beacon state and block slot=9597856`)
```
Comment (avilagaston9) [inline README.md:372]:
```suggestion
- `block_processing`: builds a fork choice store with an anchor block and state. Uses the next block available to apply `on_block`, `on_attestation` and `on_attester_slashing` handlers. Runs these handlers 30 times. To run this, at least 2 blocks and a state must be available in the db. It also needs you to set the slot manually at the beginning of an epoch. Try it for the slot that appeared when you ran checkpoint sync (you'll see in the logs something along the lines of `[Checkpoint sync] Received beacon state and block slot=9597856`)
```

File: README.md [hunk 2]
```diff
@@ -280,55 +330,114 @@ make kurtosis.clean
...
+- `multiple_block_processing`:
```
Comment (avilagaston9) [inline README.md:373]: We should add a description for this (similar to `block_processing` but with a wider range of slots) or maybe remove the file.

File: README.md [hunk 3]
```diff
@@ -280,55 +330,114 @@ make kurtosis.clean
...
+_Note: If you want to use the `:observer` GUI and not just `etop`, you'll probably need `:wx` also set in your extra applications, there is an easy way to do this, just set the `EXTRA_APPLICATIONS` environment variable to `WX` (`export EXTRA_APPLICATIONS=WX`) before starting the node_
+
+### eFlambè
+
+When optimizing code, it might be useful to have a graphic way to determine bottlenecks in the system.
+In that case, you can use [eFlambè](https://github.com/Stratus3D/eflambe) to generate flamegraphs of specific functions.
+The following code will capture information from 10 calls to `Handlers.on_block/2`, dumping it in different files named \<timestamp\>-eflambe-output.bggg.
+
+```elixir
+:eflambe.capture({LambdaEthereumConsensus.ForkChoice.Handlers, :has_block?, 2}, 10)
```
Comment (avilagaston9) [inline README.md:432]:
```suggestion
:eflambe.capture({LambdaEthereumConsensus.ForkChoice.Handlers, :on_block?, 2}, 10)
```

**Review by avilagaston9 — APPROVED** [review]
LGTM! Just a few minor comments.

File: README.md (hunk 2 above)
Comment (Arkenan) [reply README.md]: Missed it, thanks for the comment!

**Pull request merged**